# Configuration file for LE-VINS

# 结果输出路径
outputpath: ""
is_make_outputdir: true

# 是否开启可视化
is_use_visualization: true

# ROS接口配置
ros:
    imu_topic: "/imu0"
    image_topic: "/cam0"
    lidar_topic: "/livox/lidar"
    use_compressed_image: false # 使用压缩图像

    # 读取ROS包
    is_read_bag: true
    bag_file: ""

# IMU噪声建模参数
imu:
    imudatarate: 200 # IMU原始数据频率, Hz
    is_use_zupt: true # 使用零速检测和零速约束

    arw: 0.1 # deg/sqrt(hr)
    vrw: 0.1 # m/s/sqrt(hr)
    gbstd: 50.0 # deg/hr
    abstd: 50.0 # mGal
    corrtime: 1.0 # hr

    gb_prior_std: 1500 # deg/hr
    ab_prior_std: 1500 # mGal

# 优化器
optimizer:
    optimize_reprojection_std: 1.5 # 像素重投影误差
    optimize_point_to_plane_std: 0.1 # 点到面观测协方差, m
    optimize_window_size: 10 # 滑动窗口大小

    optimize_estimate_cam_extrinsic: true # 是否估计相机和IMU的外参
    optimize_estimate_cam_td: true # 是否估计相机和IMU之间的时间间隔
    optimize_cam_extrinsic_accurate: true # 相机和IMU的外参初值准确

visual:
    is_use_lidar_depth: true # 使用激光深度增强
    track_max_features: 150 # 最大提取特征数量

    # 内参 [fx, fy, cx, cy(, skew)]
    intrinsic: [787.1611861559479, 787.3928431375225, 664.4061078354368, 519.5129292754456]

    # 畸变参数 [k1, k2, p1, p2(, k3)]
    distortion: [-0.0917403092279957, 0.08134715036932794, 0.00017620136958692255,
        0.00016737385248865412]

    # 图像分辨率
    resolution: [1278, 1022]

    # 相机IMU外参 Pb = q_b_c * Pc + t_b_c
    # q (x, y, z, w)
    q_b_c: [0.498387, 0.501826, 0.500863, 0.498916]
    t_b_c: [0.076, -0.017, 0.135]

    # IMU和相机时间延时
    # t_b = t_c + td
    td_b_c: 0.0

    # 激光相机外参 Pc = q_c_l * Pl + t_c_l
    # q (x, y, z, w)
    q_c_l: [0.510336, -0.488986, 0.493321, 0.507034]
    t_c_l: [-0.00205198, -0.0897518, 0.0223906]

    # t_b = t_l + td
    td_b_l: 0.0

lidar:
    # 激光参数
    lidar_type: 1 # Livox: 1; Velodyne: 2; Ouster: 3; Hesai: 4
    scan_line: 1 # 扫描线数
    nearest_distance: 1 # 扫描盲区, m
    farthest_distance: 250.0 # 最远距离, m
    frame_rate: 10 # 帧率, Hz